apps/src/templates/sectionProgress/SectionProgressNameCell.jsx

```diff
@@ -12,7 +13,23 @@ class SectionProgressNameCell extends Component {
     sectionId: PropTypes.number.isRequired,
 
     // Provided by redux.
-    scriptName: PropTypes.string
+    scriptName: PropTypes.string,
+    scriptId: PropTypes.number,
+    userId: PropTypes.number
+  };
+
+  recordStudentNameClick = () => {
+    firehoseClient.putRecord({
+      study: 'teacher_dashboard_actions',
+      study_group: 'progress',
+      event: 'go_to_student',
+      user_id: this.props.userId,
```

Comment:
firehose.js already has an `includeUserId` flag, which tells it to look for the user id in state.pageConstants, which is only defined in code studio apps (artist, applab, etc). I would recommend augmenting the code which handles that flag to also look for that state in the current user redux. then your components which call firehoseClient.putRecord won't have to know anything about the user id themselves.